proposals/0000-android-12-issues.md

```diff
@@ -0,0 +1,277 @@
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && !AndroidTools.areBtPermissionsGranted(this, this.getPackageName())) {
+        //If Connected over USB start RouterService
+        if (isConnectedOverUSB) {
+            //Delay starting bluetoothTransport
+            waitingForBTRuntimePermissions = true;
+            btPermissionsHandler = new Handler(Looper.myLooper());
+            //Continuously Check for the Bluetooth Permissions
+            btPermissionsRunnable = new Runnable() {
+                @Override
+                public void run() {
+                    if (!AndroidTools.areBtPermissionsGranted(SdlRouterService.this, SdlRouterService.this.getPackageName())) {
+                        btPermissionsHandler.postDelayed(btPermissionsRunnable, BT_PERMISSIONS_CHECK_FREQUENCY);
+                    } else {
+                        waitingForBTRuntimePermissions = false;
+                        initBluetoothSerialService();
+                    }
+                }
+            };
+            btPermissionsHandler.postDelayed(btPermissionsRunnable, BT_PERMISSIONS_CHECK_FREQUENCY);
+            //Present Notification to take user to permissions page for the app
+            showBTPermissionsNotification();
+        } else {
+            //If connected over bluetooth and permissions are not granted, fail the initCheck
+            return false;
+        }
+    }
+
+    .....
+
+    return true;
+}
+~~~
+
+### AndroidManifest Exported Flag
+
+Starting in Android 12 any activities, services, or broadcast receivers that use intent filters will need to explicitly declare the `android:exported` attribute for the given app components. The `SdlRouterService` and `SdlReceiver` should already have the exported attribute defined and set to true. But The `USBAccessoryAttachmentActivity` will now also require this attribute to be set. Any activity that had an `intent-filter` would have a default exported value of true. Now we need to explicitly set it.
+
+~~~ xml
+<!--AndroidManifest.xml-->
+<activity
+    android:name="com.smartdevicelink.transport.USBAccessoryAttachmentActivity"
+    android:exported="true" <!--New Addition-->
+    android:launchMode="singleTop">
+    <intent-filter>
+        <action android:name="android.hardware.usb.action.USB_ACCESSORY_ATTACHED" />
+    </intent-filter>
+    .....
+</activity>
+
+~~~
+
+### PendingIntent Mutable Flag
+
+In Android 12, you must specify the mutability of each PendingIntent object that your app creates. This will impact the notifications that the `SdlRouterService` is trying to display, As we do not need to update these intents at any point we can flag them with `FLAG_IMMUTABLE`.
+
+~~~ java
+//SdlRouterService.java
+int flag = android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.S ? PendingIntent.FLAG_IMMUTABLE : 0;
+PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, intent, flag);
+
+~~~
+
+### Service Notification Delays
```

Comment:
I don't see this as a downside as it could match what we have and even be better if we dont care about the notification